server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -292,39 +286,19 @@ public  T loadLatestState(Logger logger, NamedXContentRegistry namedXContentRegi
                 }
             }
         }
-        final List<Throwable> exceptions = new ArrayList<>();
-        T state = null;
         // NOTE: we might have multiple version of the latest state if there are multiple data dirs.. for this case
-        //       we iterate only over the ones with the max version. If we have at least one state file that uses the
-        //       new format (ie. legacy == false) then we know that the latest version state ought to use this new format.
-        //       In case the state file with the latest version does not use the new format while older state files do,
-        //       the list below will be empty and loading the state will fail
+        //       we iterate only over the ones with the max version.
+        long finalMaxStateId = maxStateId;
         Collection<PathAndStateId> pathAndStateIds = files
                 .stream()
-                .filter(new StateIdAndLegacyPredicate(maxStateId, maxStateIdIsLegacy))
+                .filter(pathAndStateId -> pathAndStateId.id == finalMaxStateId)
                 .collect(Collectors.toCollection(ArrayList::new));
 
+        final List<Throwable> exceptions = new ArrayList<>();
         for (PathAndStateId pathAndStateId : pathAndStateIds) {
             try {
-                final Path stateFile = pathAndStateId.file;
-                final long id = pathAndStateId.id;
-                if (pathAndStateId.legacy) { // read the legacy format -- plain XContent
-                    final byte[] data = Files.readAllBytes(stateFile);
-                    if (data.length == 0) {
-                        logger.debug("{}: no data for [{}], ignoring...", prefix, stateFile.toAbsolutePath());
-                        continue;
-                    }
-                    try (XContentParser parser = XContentHelper
-                            .createParser(namedXContentRegistry, LoggingDeprecationHandler.INSTANCE, new BytesArray(data))) {
-                        state = fromXContent(parser);
-                    }
-                    if (state == null) {
-                        logger.debug("{}: no data for [{}], ignoring...", prefix, stateFile.toAbsolutePath());
```

Comment:
Except we didn't ignore it, we returned `null` straight away...

Reply:
details ;)